### ISSUE TITLE
Ability to set explicit proxy settings in Client & Server SSL profiles

### ISSUE THREAD

**Issue by LeonardosGitHub** [F5Networks/f5-appsvcs-extension#199]
##### Do you already have an issue opened with F5 support?
Github Issues are consistently monitored by F5 staff, but should be considered as best effort only and you should not expect to receive the same level of response as provided by F5 Support. Please [open an case](https://support.f5.com/csp/article/K2633) with F5 if this is a critical issue.

##### ISSUE TYPE

 - Feature Idea

##### AS3 BUILD/ VERSION

```
f5-appsvcs-3.15.0-6.noarch
```

##### BIGIP VERSION

```
15.0.1
```

##### SUMMARY

```
Would like AS3 to provide the ability to set the explicit proxy settings in Client & Server SSL profiles on the BIG-IP.
```

##### STEPS TO REPRODUCE

```
Does not exist today
```

##### EXPECTED RESULTS

```
CLIENT SSL PROFILE:
ltm profile client-ssl SSL_FORWARD_PROXY {
    app-service none
    **cert-key-chain {
        SSL_FWD_ExProxy {
            cert SSL_FORWARD_PROXY.crt
            key SSL_FORWARD_PROXY.key
            usage CA
        }
        default {
            cert default.crt
            key default.key
        }
    }**
    cert-lookup-by-ipaddr-port disabled
    defaults-from clientssl
    inherit-ca-certkeychain false
    inherit-certkeychain false
    mode enabled
    **ssl-forward-proxy enabled**
}

SERVER SSL PROFILE EXAMPLE:
ltm profile server-ssl SSL_FORWARD_PROXY {
    app-service none
    defaults-from serverssl
    **ssl-forward-proxy enabled**
}
```

##### ACTUAL RESULTS

```
Does not exist today
```

**Comment by dstokesf5** [comment]
Just to make sure I have this request correct, you want to be able to configure the `cert-key-chain` and `ssl-forward-proxy` settings on `ltm profile client-ssl` objects, and configure the `ssl-forward-proxy` setting `ltm profile server-ssl` objects? Are there any additional settings you need exposed in order to configure explicit proxy settings for your use case?

**Comment by LeonardosGitHub** [comment]
@dstokesf5 
At the moment there aren't any additional settings I need exposed but if the team has cycles to include others that would be great in case they are needed in the near future.

Thanks!

**Comment by dstokesf5** [comment]
This feature request has been added to our internal product backlog as AUTOTOOL-1081.

**Comment by tizianorosato-gr** [comment]
I'm interested on "Forward Proxy SSL" config too.
There are other options , othen than the cert-key-chain, that we use please see below:

(from ver 14.1.x)
ltm profile client-ssl /Common/SSL-CLIENT {
    app-service none
    cert-key-chain {
        default {
            cert /Common/default.crt
            key /Common/default.key
        }
        IssuingCA {
            cert /Common/zzzz
            chain /Common/zzzz
            key /Common/zzzz
            usage CA
        }
    }
***    cert-lookup-by-ipaddr-port enabled
    defaults-from /Common/clientssl
    inherit-ca-certkeychain false
    inherit-certkeychain true
***    ssl-forward-proxy enabled
***    ssl-forward-proxy-bypass enabled
}



ltm profile server-ssl /Common/SSL-SERVER {
    alert-timeout indefinite
    app-service none
    cache-size 262144
    cache-timeout 3600
    cipher-group none
    ciphers DEFAULT
    defaults-from /Common/serverssl
    generic-alert enabled
    handshake-timeout 10
    max-active-handshakes indefinite
    mod-ssl-methods disabled
    mode enabled
    renegotiate-period indefinite
    renegotiate-size indefinite
    renegotiation enabled
    secure-renegotiation require-strict
    server-name none
    session-mirroring disabled
    session-ticket disabled
    sni-default false
    sni-require false
***    ssl-forward-proxy enabled
***    ssl-forward-proxy-bypass enabled
    ssl-sign-hash any
    strict-resume disabled
    unclean-shutdown enabled
}

**Comment by dstokesf5** [comment]
This has been resolved as part of the 3.19.0 release.
